packages/twenty-server/src/engine/core-modules/ask-ai/ask-ai.service.ts

```diff
@@ -0,0 +1,181 @@
+import { Injectable } from '@nestjs/common';
+
+import { SqlDatabase } from 'langchain/sql_db';
+import { RunnableSequence, RunnableFunc } from '@langchain/core/runnables';
+import { StringOutputParser } from '@langchain/core/output_parsers';
+import { isUUID } from 'class-validator';
+import { DataSource } from 'typeorm';
+import omit from 'lodash.omit';
+
+import { LLMPromptTemplateEnvVar } from 'src/engine/integrations/llm-prompt-template/interfaces/llm-prompt-template-name.interface';
+
+import {
+  AskAIQueryResult,
+  RecordMetadataById,
+} from 'src/engine/core-modules/ask-ai/dtos/ask-ai-query-result.dto';
+import { WorkspaceDataSourceService } from 'src/engine/workspace-datasource/workspace-datasource.service';
+import { WorkspaceQueryRunnerService } from 'src/engine/api/graphql/workspace-query-runner/workspace-query-runner.service';
+import { LLMPromptTemplateService } from 'src/engine/integrations/llm-prompt-template/llm-prompt-template.service';
+import { LLMChatModelService } from 'src/engine/integrations/llm-chat-model/llm-chat-model.service';
+import { LLMTracingService } from 'src/engine/integrations/llm-tracing/llm-tracing.service';
+import { ObjectMetadataService } from 'src/engine/metadata-modules/object-metadata/object-metadata.service';
+import { computeTableName } from 'src/engine/utils/compute-table-name.util';
+
+@Injectable()
+export class AskAIService {
+  constructor(
+    private readonly workspaceDataSourceService: WorkspaceDataSourceService,
+    private readonly workspaceQueryRunnerService: WorkspaceQueryRunnerService,
+    private readonly llmChatModelService: LLMChatModelService,
+    private readonly llmPromptTemplateService: LLMPromptTemplateService,
+    private readonly llmTracingService: LLMTracingService,
+    private readonly objectMetadataService: ObjectMetadataService,
+  ) {}
+
+  private async getRecordMetadataById(
```

Comment:
Metadata is probably a bit misleading. We're getting additional data for UI/display purposes but we're not really getting "metadata" which is a different thing